### ISSUE TITLE
Unable to get mining info from wallet

### ISSUE THREAD

**Issue by aitorserra** [de-luxe/burstcoin-jminer#35]
Hello,

I'm getting this error. I searched at the forums and can't find a solution. I tried to increase the timeout to 120 seconds. Also to disable the walletServer=

It's a bug or a misconfiguration? thank you.

**Comment by k06a** [comment]
My configuration:
```
walletServer=http://127.0.0.1:8125
```
My check:
```
http://127.0.0.1:8125/burst?requestType=getMiningInfo
```
=>
```
{ ** Some JSON without any errors ** }
```

Why miner is unable to get mining info?

**Comment by aitorserra** [comment]
I will try that, thank you. I was using the online one.

**Comment by k06a** [comment]
@anonimo32 I have the same problem and trying to find solution too :)

**Comment by aitorserra** [comment]
ah ok! I also tried leaving this field empty as the configuration suggest but nothing.

**Comment by k06a** [comment]
@de-luxe how to build app from sources in terminal? Can you give us a command, I'll fix a few lines of code to print exception and this will help us to solve this issue.

**Comment by k06a** [comment]
Just found the solution, `127.0.0.1` and `localhost` not works, but computer network name works fine!

**Comment by de-luxe** [comment]
@k06a @anonimo32 walletServer is for getting winner and other info (optional) ... getMiningInfo is requested from poolServer / soloServer ... try to check them too, via url.

https://github.com/de-luxe/burstcoin-jminer#poolserver-required-for-pool
https://github.com/de-luxe/burstcoin-jminer#soloserver-defaulthttplocalhost8125

**Comment by de-luxe** [comment]
@k06a you can build via maven ... so you first need to install apache maven, than you only need to run:
`mvn package`
https://maven.apache.org/guides/getting-started/maven-in-five-minutes.html

on package sourcecode will be compiled and also distribution will be created within a new /dist folder

**Comment by aitorserra** [comment]
I had this:

poolServer=http://pool.burstcoin.space:8124
walletServer=https://wallet.burst-team.us:8125

And changed to this:

walletServer=windows_pc

With the local wallet running, at the moment the error does not come up.

**Comment by de-luxe** [comment]
@anonimo32 using https for walletServer may not work, thats a bug i have to fix ... try use only http wallet, like http://localhost:8125 ... no https needed there only public request made. But sure i should make that work.

**Comment by k06a** [comment]
And the last chance in case previous not works:
```
walletServer=http://windows_pc:8125
```

**Comment by k06a** [comment]
Looks like this solution just helped to one more miner: https://www.burstnation.com/wbb/index.php?thread/2085-jminer-unable-to-get-mining-info-from-wallet/&postID=38900#post38900

**Comment by de-luxe** [comment]
http://windows_pc:8125 is not a url, it is just invalid and therefore ignored i guess ... same like leaving walletServer= empty. Or am i wrong and in windows 10 whatever this is a redirect to localhost / 127.0.0.1 ?!

**Comment by aitorserra** [comment]
I changed to http://localhost:8125 and now I can see less errors. As I'm  downloading the blockchain again it says that the wallet it's out of sync so let's see what happend when it's downloaded.

Anyways, thank you very much for your help and work. The miner it's faster than the cpu one.

**Comment by k06a** [comment]
@de-luxe for me works:
```
walletServer=http://mm:8125
```
Where `mm` in network name of my MacMini (DNS name and WINS name)

**Comment by Aist5** [comment]
Helped the pool change.
Was http://burst.lexitoshi.uk:80
(Http://burst.lexitoshi.uk/howto.html)
Became http://burst.mininghere.com

**Comment by jimenezfer** [comment]
Hello, I want to use a web wallet.  Am using A rumanian pool , but still throws that error.
Here are my current settings:

numericAccountId=3xxxxxxxxxxxxxxxxxxxxxxx
poolServer=https://pool.burstcoin.ro:8080

# Winner
walletServer=https://pool.burstcoin.ro:8080
winnerRetriesOnAsync=
winnerRetryIntervalInMs=


Any Ideas ?

Thanks!

**Comment by jebuscraisp** [comment]
Same error here, anybody with the correct setting for web wallet:
Here my settings jminer.conf :

numericAccountId=3xxxxxxxxxxxxxxxxxxxxxxx


poolServer=https://pool.burstcoin.ro:8080
Winner

walletServer=https://pool.burstcoin.ro:8080


winnerRetriesOnAsync=


winnerRetryIntervalInMs=

**Comment by de-luxe** [comment]
@jebuscraisp 

The API of a pool is just a subset of the wallet api, thats the reason why a miner needs wallet and pool server to offer more information e.g. last winner. So do not use pool url for both of them.

But to stop the 'get mining info' issue only use http not https
poolServer=http://pool.burstcoin.ro:8080

**Comment by jimenezfer** [comment]
I been trying using wallet servers from here, http://status.burstcontrol.com:7777/network, I tried all the wallet servers that were up at the time in conjunction with the .ro pool , but still got the error, @de-luxe  what combination of wallet/pool do you normally use to mine?

Thanks for the clarification.

**Comment by de-luxe** [comment]
@jebuscraisp i use http://localhost:8125, as walletServer, so a local running wallet instance. 
'Unable to get mining info from wallet' has nothing to do with walletServer, and if you get it from time to time, it is no problem. (Requested from poolServer/soloServer) The miner asks for it every 2 sec. per default, if It succeeds 28 times within a minute and only fails 2 times there is nothing to worry about ...

**Comment by jimenezfer** [comment]
@de-luxe So this is just a normal state mining with PoC , so you are saying that eventually this will change to regular expected output, am already mining there then,there is no action to do then. Is there any advantage of using a local wallet over Web wallet ?

Thank you again for the clarification.

**Comment by jebuscraisp** [comment]
@de-luxe So this is just a normal state mining with PoC , so you are saying that eventually this will change to regular expected output, am already mining there then,there is no action to do then. Is there any advantage of using a local wallet over Web wallet ?

Thank you again for the clarification.

**Comment by de-luxe** [comment]
@jebuscraisp For mining it may be no difference using web or local as just some information is requested. But once you want to send money, using a web wallet makes the whole idea of crypt obsolete, as you have to trust a 3rd party again, The idea of crypto is not to relay on 3rd party like banks, or in this case the one hosting the online wallet.

**Comment by jimenezfer** [comment]
Roger that , switching to local.

**Comment by jimenezfer** [comment]
@de-luxe I think that there is something am doing wrong I have a continuous message , its not from time to time.. :/

**Comment by de-luxe** [comment]
@jebuscraisp You should maybe consider try another pool than, check burstcoin.cc for options.

**Comment by Filip5M** [comment]
I am getting this error when adding more than one path, while mining on my mac. plotPaths=/Volumes/drive1/plots,/Volumes/drive2 . As soon as I just leave 1 path it works again. Am i missing some other setting?

**Comment by shycats** [comment]
@de-luxe 
Hi, I'm getting this warning almost every time a new block is available to mine, and missing a lot of rounds because of it. I've been taking a look at the logs with debug=true and I think I understand what's going on.

All pools have DDoS attack protection systems, what that means is that if the server is too busy (like when a new block is ready to mine, cause it starts receiving a lot of DLs), any request is redirected to an external webpage that informs about the server being down or unresponsive. Of course jminer can't parse anything that's not a JSON (like the external webpage) and it looks like the status of the HTTP request doesn't return an error code >= 400, so jminer tries to parse it anyway. This shouldn't be a problem if the next request is successful, but for some reason even if it finally gets a JSON with new block info, it doesn't trigger a new mining process. Maybe it can't compare the previous block number with the new block number because of the parsing error?

I hope this helps to solve the problem.

**Comment by ciclopez** [comment]
Maybe one solution is to detect a redirection, or to check if the first character of the contents is not "{", as must be for a JSON.

**Comment by ciclopez** [comment]
@de-luxe maybe one solution is to detect a redirection, or to check if the first character of the contents is not "{", as must be for a JSON.

**Comment by shycats** [comment]
Ok, I downloaded the source code and compiled it myself.
The first thing to note is that for some reason now captures correctly the error code >= 400, so now I can discard it as the cause of missing rounds.
I added a line to the code to log when a new block is detected (newBlockNumber > blockNumber) and found that an exception is happening then, but only happens when using more than 1 plot path. So my problem is related to another issue: https://github.com/de-luxe/burstcoin-jminer/issues/49
